<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>GeneticMediation  • GeneticMediation</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>




<meta property="og:title" content="GeneticMediation " />
<meta property="og:image" content="https://tydarnell.github.io/GeneticMediation/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">GeneticMediation</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tydarnell/GeneticMediation/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>GeneticMediation </h1>
    </div>


<div id="geneticmediation-" class="section level1">

<p>April 26, 2020</p>
</div>
<div id="description" class="section level1">
<h1 class="hasAnchor">
<a href="#description" class="anchor"></a>DESCRIPTION</h1>
<pre><code>Package: GeneticMediation
Type: Package
Title: Genetic Causal Mediation
Version: 1.1.0
License: GPL-3
URL: https://tydarnell.github.io/GeneticMediation,
    https://github.com/tydarnell/GeneticMediation
Authors@R: person("Ty", "Darnell", role = c("aut", "cre"), email = "tydarnell@gmail.com")
Author: Ty Darnell [aut, cre]
Maintainer: Ty Darnell &lt;tydarnell@gmail.com&gt;
Description: Provides methods for conducting causal mediation
  analysis on data from the ROSMAP study and for cleaning,
  matching, and preparing the data for analysis.
Encoding: UTF-8
LazyData: true
RoxygenNote: 7.1.0
Imports: BiocManager, data.table, IRanges, mediation, readr, stats
Suggests: tidyverse</code></pre>
</div>
<div id="chr_numeric-chromosome-numeric" class="section level1">
<h1 class="hasAnchor">
<a href="#chr_numeric-chromosome-numeric" class="anchor"></a><code>chr_numeric</code>: Chromosome Numeric</h1>
<div id="description-1" class="section level2">
<h2 class="hasAnchor">
<a href="#description-1" class="anchor"></a>Description</h2>
<p>Convert chromosome names from character to numeric. Useful when sorting a dataframe by chromosome number.</p>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb2"><html><pre class="r"><span class="fu"><a href="reference/chr_numeric.html">chr_numeric</a></span>(<span class="no">Chr</span>)</pre></html></div>
</div>
<div id="arguments" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>Chr</code></td>
<td>a character column or vector of chromosome names</td>
</tr></tbody>
</table>
</div>
</div>
<div id="clean_data-clean-data" class="section level1">
<h1 class="hasAnchor">
<a href="#clean_data-clean-data" class="anchor"></a><code>clean_data</code>: Clean Data</h1>
<div id="description-2" class="section level2">
<h2 class="hasAnchor">
<a href="#description-2" class="anchor"></a>Description</h2>
<p>Clean snp.info, peak.info data and save as data/chipseq.RData</p>
</div>
<div id="usage-1" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-1" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb3"><html><pre class="r"><span class="fu"><a href="reference/clean_data.html">clean_data</a></span>(<span class="no">snp.path</span>, <span class="no">peak.path</span>)</pre></html></div>
</div>
<div id="arguments-1" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-1" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>snp.path</code></td>
<td>path to SNP information dataframe</td>
</tr>
<tr class="even">
<td><code>peak.path</code></td>
<td>path to Peak annotation dataframe</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="combine_data_spc-combine-data-snp-peak-clinical" class="section level1">
<h1 class="hasAnchor">
<a href="#combine_data_spc-combine-data-snp-peak-clinical" class="anchor"></a><code>combine_data_spc</code>: Combine Data SNP-Peak-Clinical</h1>
<div id="description-3" class="section level2">
<h2 class="hasAnchor">
<a href="#description-3" class="anchor"></a>Description</h2>
<p>SNP-Peak-Clinical data: combine projid, SNPs, PCs, membership, age, gender, peaks, and outcome data for a chromosome</p>
</div>
<div id="usage-2" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-2" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb4"><html><pre class="r"><span class="fu"><a href="reference/combine_data_spc.html">combine_data_spc</a></span>(<span class="no">chr</span>)</pre></html></div>
</div>
<div id="arguments-2" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-2" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>chr</code></td>
<td>chromosome name</td>
</tr></tbody>
</table>
</div>
</div>
<div id="combine_data_spg-combine-data-snp-peak-gene" class="section level1">
<h1 class="hasAnchor">
<a href="#combine_data_spg-combine-data-snp-peak-gene" class="anchor"></a><code>combine_data_spg</code>: Combine Data SNP-Peak-Gene</h1>
<div id="description-4" class="section level2">
<h2 class="hasAnchor">
<a href="#description-4" class="anchor"></a>Description</h2>
<p>SNP-Peak-Gene Data: combine projid, SNPs, PCs, membership, age, gender, peaks, gene and outcome data for a chromosome.</p>
</div>
<div id="usage-3" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-3" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb5"><html><pre class="r"><span class="fu"><a href="reference/combine_data_spg.html">combine_data_spg</a></span>(<span class="no">chr</span>)</pre></html></div>
</div>
<div id="arguments-3" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-3" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>chr</code></td>
<td>chromosome name</td>
</tr></tbody>
</table>
</div>
</div>
<div id="geneticmediation-geneticmediation-a-package-for-conducting-causal-mediation-analysis-on-rosmap-data" class="section level1">
<h1 class="hasAnchor">
<a href="#geneticmediation-geneticmediation-a-package-for-conducting-causal-mediation-analysis-on-rosmap-data" class="anchor"></a><code>GeneticMediation</code>: GeneticMediation: A package for conducting causal mediation analysis on ROSMAP data</h1>
<div id="description-5" class="section level2">
<h2 class="hasAnchor">
<a href="#description-5" class="anchor"></a>Description</h2>
<p>GeneticMediation provides methods for conducting causal mediation analysis on data from the ROSMAP study. It also provides methods for cleaning, matching, and preparing the data for analysis.</p>
</div>
<div id="seealso" class="section level2">
<h2 class="hasAnchor">
<a href="#seealso" class="anchor"></a>Seealso</h2>
<p>Useful links:</p>
<ul>
<li><p><a href="https://tydarnell.github.io/GeneticMediation">https://tydarnell.github.io/GeneticMediation</a></p></li>
<li><p><a href="https://github.com/tydarnell/GeneticMediation">https://github.com/tydarnell/GeneticMediation</a></p></li>
</ul>
</div>
<div id="author" class="section level2">
<h2 class="hasAnchor">
<a href="#author" class="anchor"></a>Author</h2>
<p>Maintainer : Ty Darnell <a href="mailto:tydarnell@gmail.com" class="email">tydarnell@gmail.com</a></p>
</div>
</div>
<div id="last_to_first-last-to-first" class="section level1">
<h1 class="hasAnchor">
<a href="#last_to_first-last-to-first" class="anchor"></a><code>last_to_first</code>: Last to First</h1>
<div id="description-6" class="section level2">
<h2 class="hasAnchor">
<a href="#description-6" class="anchor"></a>Description</h2>
<p>Make the last column the first column in a dataframe</p>
</div>
<div id="usage-4" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-4" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb6"><html><pre class="r"><span class="fu"><a href="reference/last_to_first.html">last_to_first</a></span>(<span class="no">df</span>)</pre></html></div>
</div>
<div id="arguments-4" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-4" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>df</code></td>
<td>dataframe</td>
</tr></tbody>
</table>
</div>
</div>
<div id="lu-length-unique" class="section level1">
<h1 class="hasAnchor">
<a href="#lu-length-unique" class="anchor"></a><code>lu</code>: Length Unique</h1>
<div id="description-7" class="section level2">
<h2 class="hasAnchor">
<a href="#description-7" class="anchor"></a>Description</h2>
<p>Get the length of unique values in a vector</p>
</div>
<div id="usage-5" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-5" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb7"><html><pre class="r"><span class="fu"><a href="reference/lu.html">lu</a></span>(<span class="no">x</span>)</pre></html></div>
</div>
<div id="arguments-5" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-5" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>x</code></td>
<td>vector</td>
</tr></tbody>
</table>
</div>
</div>
<div id="make_folder-make-folder" class="section level1">
<h1 class="hasAnchor">
<a href="#make_folder-make-folder" class="anchor"></a><code>make_folder</code>: Make Folder</h1>
<div id="description-8" class="section level2">
<h2 class="hasAnchor">
<a href="#description-8" class="anchor"></a>Description</h2>
<p>Make a folder only if the folder does not already exist</p>
</div>
<div id="usage-6" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-6" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb8"><html><pre class="r"><span class="fu"><a href="reference/make_folder.html">make_folder</a></span>(<span class="no">path</span>)</pre></html></div>
</div>
<div id="arguments-6" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-6" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>path</code></td>
<td>folder path</td>
</tr></tbody>
</table>
</div>
</div>
<div id="match_all-match-snps-peaks-all-chromosomes" class="section level1">
<h1 class="hasAnchor">
<a href="#match_all-match-snps-peaks-all-chromosomes" class="anchor"></a><code>match_all</code>: Match SNPs Peaks All Chromosomes</h1>
<div id="description-9" class="section level2">
<h2 class="hasAnchor">
<a href="#description-9" class="anchor"></a>Description</h2>
<p>Match SNPs and Peaks in all chromosomes and return dataframe of matches</p>
</div>
<div id="usage-7" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-7" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb9"><html><pre class="r"><span class="fu"><a href="reference/match_all.html">match_all</a></span>(<span class="no">snp.info</span>, <span class="no">peak.info</span>, <span class="no">chrs</span>)</pre></html></div>
</div>
<div id="arguments-7" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-7" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>snp.info</code></td>
<td>SNP information</td>
</tr>
<tr class="even">
<td><code>peak.info</code></td>
<td>Peak information</td>
</tr>
<tr class="odd">
<td><code>chrs</code></td>
<td>character vector of chromosome names</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="match_snp_peak-match-snps-and-peaks" class="section level1">
<h1 class="hasAnchor">
<a href="#match_snp_peak-match-snps-and-peaks" class="anchor"></a><code>match_snp_peak</code>: Match SNPs and Peaks</h1>
<div id="description-10" class="section level2">
<h2 class="hasAnchor">
<a href="#description-10" class="anchor"></a>Description</h2>
<p>Match SNPs and Peaks in a chromosome</p>
</div>
<div id="usage-8" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-8" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb10"><html><pre class="r"><span class="fu"><a href="reference/match_snp_peak.html">match_snp_peak</a></span>(<span class="no">snp.info</span>, <span class="no">peak.info</span>)</pre></html></div>
</div>
<div id="arguments-8" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-8" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>snp.info</code></td>
<td>SNP information for chr</td>
</tr>
<tr class="even">
<td><code>peak.info</code></td>
<td>Peak information chr</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="medres_spc-snp-peak-clinical-mediation-data-prep" class="section level1">
<h1 class="hasAnchor">
<a href="#medres_spc-snp-peak-clinical-mediation-data-prep" class="anchor"></a><code>med.res_spc</code>: SNP-Peak-Clinical Mediation Data Prep</h1>
<div id="description-11" class="section level2">
<h2 class="hasAnchor">
<a href="#description-11" class="anchor"></a>Description</h2>
<p>Prepare a chromosome of SNP-Peak-Clinical data for mediation analysis and save as “data/spc.res/res_ <code>chr</code> .RData”</p>
</div>
<div id="usage-9" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-9" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb11"><html><pre class="r"><span class="fu"><a href="reference/med.res_spc.html">med.res_spc</a></span>(<span class="no">chr</span>, <span class="no">all_matches</span>)</pre></html></div>
</div>
<div id="arguments-9" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-9" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>chr</code></td>
<td>chromosome name</td>
</tr>
<tr class="even">
<td><code>all_matches</code></td>
<td>dataframe of SNP-Peak matches for all chromosomes</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="medres_spg-snp-peak-gene-mediation-data-prep" class="section level1">
<h1 class="hasAnchor">
<a href="#medres_spg-snp-peak-gene-mediation-data-prep" class="anchor"></a><code>med.res_spg</code>: SNP-Peak-Gene Mediation Data Prep</h1>
<div id="description-12" class="section level2">
<h2 class="hasAnchor">
<a href="#description-12" class="anchor"></a>Description</h2>
<p>Prepare a chromosome SNP-Peak-Gene data for mediation analysis data and save as “data/spg.res/res_ <code>chr</code> .RData”</p>
</div>
<div id="usage-10" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-10" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb12"><html><pre class="r"><span class="fu"><a href="reference/med.res_spg.html">med.res_spg</a></span>(<span class="no">chr</span>, <span class="no">med.data</span>, <span class="no">matches</span>, <span class="no">mediator.path</span>)</pre></html></div>
</div>
<div id="arguments-10" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-10" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>chr</code></td>
<td>chromosome name</td>
</tr>
<tr class="even">
<td><code>med.data</code></td>
<td>gene mediation data</td>
</tr>
<tr class="odd">
<td><code>matches</code></td>
<td>SNP-Peak-Gene matches dataframe</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="med_all_spc-snp-peak-clinical-mediation-table" class="section level1">
<h1 class="hasAnchor">
<a href="#med_all_spc-snp-peak-clinical-mediation-table" class="anchor"></a><code>med_all_spc</code>: SNP-Peak-Clinical Mediation Table</h1>
<div id="description-13" class="section level2">
<h2 class="hasAnchor">
<a href="#description-13" class="anchor"></a>Description</h2>
<p>Create a table of SNP-Peak-Clinical mediation results for all chromosomes</p>
</div>
<div id="usage-11" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-11" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb13"><html><pre class="r"><span class="fu"><a href="reference/med_all_spc.html">med_all_spc</a></span>(<span class="no">chrs</span>, <span class="no">covar</span>, <span class="no">simulations</span>)</pre></html></div>
</div>
<div id="arguments-11" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-11" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>chrs</code></td>
<td>character vector of chromosome names</td>
</tr>
<tr class="even">
<td><code>covar</code></td>
<td>covariates string, each covariate separated by +</td>
</tr>
<tr class="odd">
<td><code>simulations</code></td>
<td>number of simulations to run</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="med_chr_spc-snp-peak-clinical-mediation-table-chromosome" class="section level1">
<h1 class="hasAnchor">
<a href="#med_chr_spc-snp-peak-clinical-mediation-table-chromosome" class="anchor"></a><code>med_chr_spc</code>: SNP-Peak-Clinical Mediation Table Chromosome</h1>
<div id="description-14" class="section level2">
<h2 class="hasAnchor">
<a href="#description-14" class="anchor"></a>Description</h2>
<p>Create SNP-Peak-Clinical mediation table for SNP-Peak matches in a chromosome</p>
</div>
<div id="usage-12" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-12" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb14"><html><pre class="r"><span class="fu"><a href="reference/med_chr_spc.html">med_chr_spc</a></span>(<span class="no">matches</span>, <span class="no">med.res.data</span>, <span class="no">covar</span>, <span class="no">simulations</span>)</pre></html></div>
</div>
<div id="arguments-12" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-12" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>matches</code></td>
<td>SNP-Peak matches dataframe for a chromosome</td>
</tr>
<tr class="even">
<td><code>med.res.data</code></td>
<td>mediation data</td>
</tr>
<tr class="odd">
<td><code>covar</code></td>
<td>covariates string, separate covariates with +</td>
</tr>
<tr class="even">
<td><code>simulations</code></td>
<td>number of simulations to run</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="med_chr_spg-snp-peak-gene-mediation-table-chromosome" class="section level1">
<h1 class="hasAnchor">
<a href="#med_chr_spg-snp-peak-gene-mediation-table-chromosome" class="anchor"></a><code>med_chr_spg</code>: SNP-Peak-Gene Mediation Table Chromosome</h1>
<div id="description-15" class="section level2">
<h2 class="hasAnchor">
<a href="#description-15" class="anchor"></a>Description</h2>
<p>Create a SNP-Peak-Gene mediation table for a chromosome</p>
</div>
<div id="usage-13" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-13" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb15"><html><pre class="r"><span class="fu"><a href="reference/med_chr_spg.html">med_chr_spg</a></span>(<span class="no">chr</span>, <span class="no">gene_matches</span>, <span class="no">simulations</span>)</pre></html></div>
</div>
<div id="arguments-13" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-13" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>chr</code></td>
<td>chromosome name</td>
</tr>
<tr class="even">
<td><code>gene_matches</code></td>
<td>list of peak-SNP matches for each gene</td>
</tr>
<tr class="odd">
<td><code>simulations</code></td>
<td>number of simulations to run</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="med_table_spc-snp-peak-clinical-mediation-table" class="section level1">
<h1 class="hasAnchor">
<a href="#med_table_spc-snp-peak-clinical-mediation-table" class="anchor"></a><code>med_table_spc</code>: SNP-Peak-Clinical Mediation Table</h1>
<div id="description-16" class="section level2">
<h2 class="hasAnchor">
<a href="#description-16" class="anchor"></a>Description</h2>
<p>Create SNP-Peak-Clinical mediation table for a SNP-Peak match</p>
</div>
<div id="usage-14" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-14" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb16"><html><pre class="r"><span class="fu"><a href="reference/med_table_spc.html">med_table_spc</a></span>(<span class="no">med.res.data</span>, <span class="no">match_row</span>, <span class="no">covar</span>, <span class="no">simulations</span>)</pre></html></div>
</div>
<div id="arguments-14" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-14" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>med.res.data</code></td>
<td>mediation data</td>
</tr>
<tr class="even">
<td><code>match_row</code></td>
<td>row in match dataframe, should have 2 columns</td>
</tr>
<tr class="odd">
<td><code>covar</code></td>
<td>covariates character variable, separate covariates with +</td>
</tr>
<tr class="even">
<td><code>simulations</code></td>
<td>number of simulations to run</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="med_table_spg-snp-peak-gene-mediation-table" class="section level1">
<h1 class="hasAnchor">
<a href="#med_table_spg-snp-peak-gene-mediation-table" class="anchor"></a><code>med_table_spg</code>: SNP-Peak-Gene Mediation Table</h1>
<div id="description-17" class="section level2">
<h2 class="hasAnchor">
<a href="#description-17" class="anchor"></a>Description</h2>
<p>Create a mediation table for one SNP-Peak-Gene match</p>
</div>
<div id="usage-15" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-15" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb17"><html><pre class="r"><span class="fu"><a href="reference/med_table_spg.html">med_table_spg</a></span>(<span class="no">med_dat</span>, <span class="no">match_row</span>, <span class="no">simulations</span>)</pre></html></div>
</div>
<div id="arguments-15" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-15" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>med_dat</code></td>
<td>gene mediation data</td>
</tr>
<tr class="even">
<td><code>match_row</code></td>
<td>row in match dataframe: col1 SNP, col2 Peak, col3 Gene</td>
</tr>
<tr class="odd">
<td><code>simulations</code></td>
<td>number of simulations to run</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="snp_peak_bychr-snps-and-peaks-by-chromosome" class="section level1">
<h1 class="hasAnchor">
<a href="#snp_peak_bychr-snps-and-peaks-by-chromosome" class="anchor"></a><code>snp_peak_bychr</code>: SNPs and Peaks by Chromosome</h1>
<div id="description-18" class="section level2">
<h2 class="hasAnchor">
<a href="#description-18" class="anchor"></a>Description</h2>
<p>save snp.info, peak.info for each chromosome as “data/chipseq/chipseq_ <code>chr</code> .RData”</p>
</div>
<div id="usage-16" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-16" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb18"><html><pre class="r"><span class="fu"><a href="reference/snp_peak_bychr.html">snp_peak_bychr</a></span>(<span class="no">chrs</span>, <span class="no">snps</span>, <span class="no">peaks</span>)</pre></html></div>
</div>
<div id="arguments-16" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-16" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>chrs</code></td>
<td>Character vector of chromosomes names</td>
</tr>
<tr class="even">
<td><code>snps</code></td>
<td>SNP information dataframe</td>
</tr>
<tr class="odd">
<td><code>peaks</code></td>
<td>Peak information dataframe</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sum_zero-sum-zero" class="section level1">
<h1 class="hasAnchor">
<a href="#sum_zero-sum-zero" class="anchor"></a><code>sum_zero</code>: Sum Zero</h1>
<div id="description-19" class="section level2">
<h2 class="hasAnchor">
<a href="#description-19" class="anchor"></a>Description</h2>
<p>Get names of columns that have a sum of zero</p>
</div>
<div id="usage-17" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-17" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb19"><html><pre class="r"><span class="fu"><a href="reference/sum_zero.html">sum_zero</a></span>(<span class="no">df</span>)</pre></html></div>
</div>
<div id="arguments-17" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-17" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>df</code></td>
<td>dataframe</td>
</tr></tbody>
</table>
</div>
</div>
<div id="transpose_readcount-transpose-readcount" class="section level1">
<h1 class="hasAnchor">
<a href="#transpose_readcount-transpose-readcount" class="anchor"></a><code>transpose_readcount</code>: Transpose Readcount</h1>
<div id="description-20" class="section level2">
<h2 class="hasAnchor">
<a href="#description-20" class="anchor"></a>Description</h2>
<p>Read in and transpose ChIP-seq Readcount dataframe and match project id</p>
</div>
<div id="usage-18" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-18" class="anchor"></a>Usage</h2>
<div class="sourceCode" id="cb20"><html><pre class="r"><span class="fu"><a href="reference/transpose_readcount.html">transpose_readcount</a></span>(<span class="no">readcount.path</span>)</pre></html></div>
</div>
<div id="arguments-18" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments-18" class="anchor"></a>Arguments</h2>
<table>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td><code>readcount.path</code></td>
<td>ChIP-seq readcount dataframe file path</td>
</tr></tbody>
</table>
</div>
</div>


  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by Ty Darnell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


